google_guest_agent/non_windows_accounts.go

```diff
@@ -340,25 +338,16 @@ func createGoogleUser(user string) error {
 // permissions are removed but the user remains on the system. Group membership
 // is not changed.
 func removeGoogleUser(user string) error {
-	var err error
 	if config.Section("Accounts").Key("deprovision_remove").MustBool(true) {
 		userdel := config.Section("Accounts").Key("userdel_cmd").MustString("userdel -r {user}")
```

Comment:
one thing to note is that by moving functions into per-platform variants as you have, that you're not doing so for everything. so here we had a consistency between adding and removing users that was easy to see - both built their local shell command and then executed it. now we have add user which calls a platform-dependent function, and then removeuser which is doing the local building of command. i am not sure what we gain by having createUser be a consistent name for a function

Reply:
This one was tough, I mainly moved it because these had windows comparisons and the remove one didn't (we can add it like we've done elsewhere though). One thing I want to do is ensure there is no misunderstanding in the code, the fact that unix only functions are in the same namespace as the windows build without a clear naming scheme to differentiate is worrisome.

Reply:
OK, let's discuss further but can leave for now. I think where there is a common behavior but with platform-dependent implementation, esp if one or both can only be compiled on a platform, we can do this. Otherwise just mentioning in naming is probably better and easier to manage in the long run.